src/claimtrie.cpp

```diff
@@ -1434,198 +935,158 @@ bool CClaimTrieCache::removeClaimFromTrie(const std::string& name, const COutPoi
-        // If the row exists in the base, copy its claims into the new row.
-        bool exists = base->getQueueNameRow(name, queueNameRow);
-        if (!exists)
-            if (!createIfNotExists)
-                return itQueueNameRow;
-        // Stick the new row in the cache
-        std::pair<queueNameType::iterator, bool> ret;
-        ret = claimQueueNameCache.insert(std::pair<std::string, queueNameRowType>(name, queueNameRow));
-        assert(ret.second);
-        itQueueNameRow = ret.first;
     }
-    return itQueueNameRow;
+    return true;
 }
 
-bool CClaimTrieCache::addClaim(const std::string& name, const COutPoint& outPoint, uint160 claimId, CAmount nAmount, int nHeight) const
+bool CClaimTrieCache::addClaim(const std::string& name, const COutPoint& outPoint, uint160 claimId, CAmount nAmount, int nHeight)
 {
     LogPrintf("%s: name: %s, txhash: %s, nOut: %d, claimId: %s, nAmount: %d, nHeight: %d, nCurrentHeight: %d\n", __func__, name, outPoint.hash.GetHex(), outPoint.n, claimId.GetHex(), nAmount, nHeight, nCurrentHeight);
     assert(nHeight == nCurrentHeight);
     CClaimValue currentClaim;
     int delayForClaim;
-    if (getOriginalInfoForName(name, currentClaim) && currentClaim.claimId == claimId)
-    {
+    if (getOriginalInfoForName(name, currentClaim) && currentClaim.claimId == claimId) {
         LogPrintf("%s: This is an update to a best claim.\n", __func__);
         delayForClaim = 0;
-    }
-    else
-    {
+    } else {
         delayForClaim = getDelayForName(name);
     }
     CClaimValue newClaim(outPoint, claimId, nAmount, nHeight, nHeight + delayForClaim);
     return addClaimToQueues(name, newClaim);
 }
 
-bool CClaimTrieCache::undoSpendClaim(const std::string& name, const COutPoint& outPoint, uint160 claimId, CAmount nAmount, int nHeight, int nValidAtHeight) const
+bool CClaimTrieCache::undoSpendClaim(const std::string& name, const COutPoint& outPoint, uint160 claimId, CAmount nAmount, int nHeight, int nValidAtHeight)
 {
     LogPrintf("%s: name: %s, txhash: %s, nOut: %d, claimId: %s, nAmount: %d, nHeight: %d, nValidAtHeight: %d, nCurrentHeight: %d\n", __func__, name, outPoint.hash.GetHex(), outPoint.n, claimId.GetHex(), nAmount, nHeight, nValidAtHeight, nCurrentHeight);
     CClaimValue claim(outPoint, claimId, nAmount, nHeight, nValidAtHeight);
-    if (nValidAtHeight < nCurrentHeight)
-    {
+    if (nValidAtHeight < nCurrentHeight) {
         nameOutPointType entry(name, claim.outPoint);
         addToExpirationQueue(claim.nHeight + base->nExpirationTime, entry);
         return insertClaimIntoTrie(name, claim, false);
-    }
-    else
-    {
+    } else {
         return addClaimToQueues(name, claim);
     }
 }
 
-bool CClaimTrieCache::addClaimToQueues(const std::string& name, CClaimValue& claim) const
+template <typename K, typename V>
+typename std::map<K, V>::iterator CClaimTrieCache::getQueueCacheRow(const K& key, std::map<K, V>& map, bool createIfNotExists)
```

Comment:
I think this method needs to return a read-only collection. Are people going to add to modify that data thinking that they're updating the actual queue? In looking at the code, it appears that we do return the real buffer, but that's only one of three cases. The other two: we read something off disk or we use that `createIfNotExists` flag.

Reply:
It private one, so it safe to return iterator over second argument, which is map.